bundles/org.openhab.binding.mideaac/src/main/java/org/openhab/binding/mideaac/internal/connection/ConnectionManager.java

```diff
@@ -0,0 +1,540 @@
+import org.openhab.binding.mideaac.internal.security.Decryption8370Result;
+import org.openhab.binding.mideaac.internal.security.Security;
+import org.openhab.binding.mideaac.internal.security.Security.MsgType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ConnectionManager} class is responsible for managing the state of the TCP connection to the
+ * indoor AC unit evaporator.
+ *
+ * @author Jacek Dobrowolski - Initial Contribution
+ * @author Bob Eckhoff - Revised logic to reconnect with security before each poll or command
+ * 
+ *         This gets around the issue that any command needs to be within 30 seconds of the authorization
+ *         in testing this only adds 50 ms, but allows polls at longer intervals
+ */
+@NonNullByDefault
+public class ConnectionManager {
+    private Logger logger = LoggerFactory.getLogger(ConnectionManager.class);
+
+    private final String ipAddress;
+    private final int ipPort;
+    private final int timeout;
+    private String key;
+    private String token;
+    private final String cloud;
+    private final String deviceId;
+    private Response lastResponse;
+    private CloudProviderDTO cloudProvider;
+    private Security security;
+    private final int version;
+    private final boolean promptTone;
+    private boolean deviceIsConnected;
+    private int droppedCommands = 0;
+
+    /**
+     * True allows command retry if null response
+     */
+    private boolean retry = true;
+
+    public ConnectionManager(String ipAddress, int ipPort, int timeout, String key, String token, String cloud,
+            String email, String password, String deviceId, int version, boolean promptTone) {
+        this.deviceIsConnected = false;
+        this.ipAddress = ipAddress;
+        this.ipPort = ipPort;
+        this.timeout = timeout;
+        this.key = key;
+        this.token = token;
+        this.cloud = cloud;
+        this.deviceId = deviceId;
+        this.version = version;
+        this.promptTone = promptTone;
+        this.lastResponse = new Response(HexFormat.of().parseHex("C00042667F7F003C0000046066000000000000000000F9ECDB"),
+                version, "query", (byte) 0xc0);
+        this.cloudProvider = CloudProviderDTO.getCloudProvider(cloud);
+        this.security = new Security(cloudProvider);
+    }
+
+    private Socket socket = new Socket();
+    private InputStream inputStream = new ByteArrayInputStream(new byte[0]);
+    private DataOutputStream writer = new DataOutputStream(System.out);
+
+    /**
+     * Gets last response
+     * 
+     * @return byte array of last response
+     */
+    public Response getLastResponse() {
+        return this.lastResponse;
+    }
+
+    /**
+     * Validate if String is blank
+     * 
+     * @param str string to be evaluated
+     * @return boolean true or false
+     */
+    public static boolean isBlank(String str) {
+        return str.trim().isEmpty();
+    }
```

Comment:
Remove this and use `String.isBlank()` wherever it is called.

Reply:
ok